test/libsolidity/SolidityEndToEndTest.cpp

```diff
@@ -14234,6 +14234,70 @@ BOOST_AUTO_TEST_CASE(base_access_to_function_type_variables)
 	ABI_CHECK(callContractFunction("h()"), encodeArgs(2));
 }
 
+BOOST_AUTO_TEST_CASE(code_access)
+{
+	char const* sourceCode = R"(
+		contract C {
+			function lengths() public pure returns (bool) {
+				uint crLen = type(D).creationCode.length;
+				uint runLen = type(D).runtimeCode.length;
+				require(runLen < crLen);
+				require(crLen >= 0x95 && crLen < 0xd0);
```

Comment:
Tests are still failing - probably around here. This and the next line seem quite fragile, are they really beneficial? Might it make sense, if we instead had a test that creates a contract both with ``new`` and using the ``creationCode`` with assembly and just compares that they have the same bytecode and that that bytecode is the same as ``runtimeCode``?

Reply:
Maybe for checking ``creationCode`` a bit more the constructor could set some state variables and we could verify their values in both versions match? Not sure whether that works well for ``creationCode``, but for ``runtimeCode`` it should work well...